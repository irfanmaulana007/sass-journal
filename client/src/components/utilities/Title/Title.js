import React from 'react';
import PropTypes from "prop-types"

const Title = ({ title, subTitle }) => {
    return (
        <div className="mb-5">
            <h3 className="text-center text-uppercase">{title}</h3>
            <h6 className="text-center txt-dark">{subTitle}</h6>
        </div>
    )
}

Title.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}

export default Title