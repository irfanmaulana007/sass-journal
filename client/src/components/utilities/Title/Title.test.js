import React from 'react'
import { shallow } from 'enzyme'

import Title from './Title'

const props = {
    title: 'Test Title',
    subTitle: 'This is Sub Title',
}
const wrapped = shallow(<Title title={props.title} subTitle={props.subTitle} />)

describe('Title Component', () => {
    it('should render the Title Component correctly', () => {   
        expect(wrapped).toMatchSnapshot();
    });

    it('renders the Titles Label', () => { 
        expect(wrapped.find('h3').text()).toEqual(props.title);
        expect(wrapped).toMatchSnapshot();
    });

    it('renders the Titles Sub Label', () => { 
        expect(wrapped.find('h6').text()).toEqual(props.subTitle);
        expect(wrapped).toMatchSnapshot();
    });
});