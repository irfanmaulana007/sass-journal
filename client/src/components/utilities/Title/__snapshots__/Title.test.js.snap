// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders the Titles Label 1`] = `ShallowWrapper {}`;

exports[`Title renders the Titles Sub Label 1`] = `ShallowWrapper {}`;

exports[`Title should render the Title Component correctly 1`] = `ShallowWrapper {}`;
